satellite/gracefulexit/endpoint.go

```diff
@@ -620,3 +625,39 @@ func (endpoint *Endpoint) handleFailed(ctx context.Context, pending *pendingMap,
 
 	return nil
 }
+
+func (endpoint *Endpoint) updatePointer(ctx context.Context, exitingNodeID storj.NodeID, receivingNodeID storj.NodeID, path []byte, pieceNum int32) (err error) {
+	defer mon.Task()(&ctx)(&err)
+
+	// remove the node from the pointer
+	pointer, err := endpoint.metainfo.Get(ctx, string(path))
```

Comment:
This could end up overwriting a perfectly valid piece. The reference pointer MUST be from the point where you start the transfer not an arbitrary point in time.

Reply:
New PR here: https://github.com/storj/storj/pull/3397